cmd/notification.go

```diff
@@ -890,6 +891,290 @@ func (sys *NotificationSys) CollectNetPerfInfo(size int64) map[string][]ServerNe
 	return reply
 }
 
+// NetOBDInfo - Net OBD information
+func (sys *NotificationSys) NetOBDInfo(ctx context.Context) madmin.ServerNetOBDInfo {
+	var sortedGlobalEndpoints []string
+
+	/*
+			Ensure that only untraversed links are visited by this server
+		        i.e. if netOBD tests have been performed between a -> b, then do
+			not run it between b -> a
+
+		        The graph of tests looks like this
+
+		            a   b   c   d
+		        a | o | x | x | x |
+		        b | o | o | x | x |
+		        c | o | o | o | x |
+		        d | o | o | o | o |
+
+		        'x's should be tested, and 'o's should be skipped
+	*/
+	stripPath := func(hostPath string) string {
+		return strings.Split(hostPath, "/")[0]
+	}
+
+	hostMap := map[string]bool{}
+	for _, ez := range globalEndpoints {
+		for _, e := range ez.Endpoints {
+			host := stripPath(e.Host)
```

Comment:
if you are only interested in host you can use e.Hostname() - you can avoid stripPath() function altogether